runtime/src/main/java/org/corfudb/runtime/collections/StreamContext.java

```diff
@@ -0,0 +1,268 @@
+ *
+ * Created by sneginhal on 10/22/2019.
+ */
+@Slf4j
+public class StreamContext {
+
+    /**
+     * Corfu Runtime.
+     */
+    private final CorfuRuntime runtime;
+
+    /**
+     * StreamListener.
+     */
+    @Getter
+    private final StreamListener listener;
+
+    /**
+     * Namespace of interest.
+     */
+    @Getter
+    private final String namespace;
+
+    /**
+     * Tables of interest.
+     *
+     * Map of (Stream Id - TableSchema)
+     */
+    @Getter
+    private final Map<UUID,
+        TableSchema<? extends Message, ? extends Message, ? extends Message>> tablesOfInterest;
+
+    /**
+     * Starting address.
+     */
+    private final long startAddress;
+
+    /**
+     * Last read address of the stream.
+     */
+    @Getter
+    private long lastReadAddress;
+
+    /**
+     * A flag to indicate if a TxPoller is processing this StreamContext.
+     */
+    private final AtomicBoolean txPollerInProgress = new AtomicBoolean(false);
+
+    /**
+     * Exception encountered when polling for this StreamContext.
+     */
+    @Getter
+    @Setter
+    private volatile Throwable pollerException = null;
+
+    /**
+     * Is the StreamContext ready to be reaped.
+     *
+     * After an exception has been delivered to the client, the StreamContext is
+     * no longer active and can be reaped.
+     */
+    @Getter
+    private volatile boolean readyToBeReaped = false;
+
+    /**
+     * Maximum size of the streamQueue.
+     */
+    private final static int QUEUE_CAPACITY = 512;
+
+    /**
+     * Bounded queue of CorfuStreamEntries to be delivered to the client.
+     */
+    private final LinkedBlockingQueue<CorfuStreamEntries> streamQueue =
+            new LinkedBlockingQueue<>(QUEUE_CAPACITY);
+
+    /**
+     * A flag to indicate if a notification thread is already running for this StreamContext.
+     */
+    private final AtomicBoolean notificationInProgress = new AtomicBoolean(false);
+
+    /**
+     * Constructor.
+     *
+     * @param listener The client listener.
+     * @param namespace The namespace of interest.
+     * @param tableSchemas Only updates from these tables will be returned.
+     * @param startAddress The starting address from which all updates are to be streamed.
+     */
+    public StreamContext(@Nonnull CorfuRuntime runtime, @Nonnull StreamListener listener,
+            @Nonnull String namespace, @Nonnull List<TableSchema> tableSchemas,
+            long startAddress) {
+        this.runtime = runtime;
+        this.listener = listener;
+        this.namespace = namespace;
+        this.tablesOfInterest = tableSchemas.stream().collect(Collectors.toMap(
+                        e -> runtime.getStreamID(runtime.getTableRegistry()
+                                .getFullyQualifiedTableName(namespace, e.getTableName())),
+                        e -> e));
+        this.startAddress = startAddress;
+        this.lastReadAddress = startAddress;
```

Comment:
Is startAddress included in the stream?

Reply:
No, start address is not included in the stream.